### PR TITLE
interface to new nauty generator for Hasse diagrams

### PR DESCRIPTION
**Pull request by fchapoton**
add a basic interface to the new capability of nauty, namely a generator for Hasse diagrams of posets

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: src/sage/graphs/digraph_generators.py [hunk 1]
```diff
@@ -758,6 +759,64 @@ def nauty_directg(self, graphs, options='', debug=False):
...
+          string. A line leading with ">A" indicates a successful initiation of
+          the program with some information on the arguments, while a line
+          beginning with ">E" indicates an error with the input.
+
+        The possible options, obtained as output of ``genposetg --help``::
+
+            n: the number of vertices
+            o: digraph6 output in arbitrary order
+            t: digraph6 output in topological order
+            q: supresses statistics except for the final count
```
Comment (dcoudert) [inline src/sage/graphs/digraph_generators.py]: Statistics are written in stderr and never reported by this function. 
So either, option `q` should always be used and not documented, or you should find a way to report statistics.

File: src/sage/graphs/digraph_generators.py [hunk 2]
```diff
@@ -758,6 +759,64 @@ def nauty_directg(self, graphs, options='', debug=False):
...
+        - ``debug`` -- boolean (default: ``False``); if ``True`` the first line
+          of ``genposetg``'s output to standard error is captured and the first
+          call to the generator's ``next()`` function will return this line as a
+          string. A line leading with ">A" indicates a successful initiation of
+          the program with some information on the arguments, while a line
+          beginning with ">E" indicates an error with the input.
+
+        The possible options, obtained as output of ``genposetg --help``::
+
+            n: the number of vertices
```
Comment (dcoudert) [inline src/sage/graphs/digraph_generators.py]: add that `n=0..16`

**Review by dcoudert — CHANGES_REQUESTED** [review]

**Review by dcoudert — APPROVED** [review]
LGTM.

**Comment by fchapoton** [comment]
merci, David !

**Comment by fchapoton** [comment]
une typo dans le doc, je repasse en positif

**Pull request merged**